api/src/main/java/org/openmrs/api/db/hibernate/HibernateDiagnosisDAO.java

```diff
@@ -86,32 +90,53 @@ public Diagnosis saveDiagnosis(Diagnosis diagnosis) {
 	}
 
 	/**
-	 * Gets all diagnoses for a given encounter
-	 *
-	 * @param encounter the specific encounter to get the diagnoses for.
-	 * @return list of diagnoses for an encounter
+	 * @see org.openmrs.api.db.DiagnosisDAO#getDiagnosesByEncounter(Encounter, boolean, boolean)
 	 */
 	@Override
-	public List<Diagnosis> getDiagnoses(Encounter encounter){
-		Query query = sessionFactory.getCurrentSession().createQuery(
-			"from Diagnosis d where d.encounter.encounterId = :encounterId order by dateCreated desc");
-		query.setInteger("encounterId", encounter.getId());
-		return query.list();	
+	public List<Diagnosis> getDiagnosesByEncounter(Encounter encounter, boolean primaryOnly, boolean confirmedOnly) {
+		String queryString = "from Diagnosis d where d.encounter.id = :encounterId";
```

Comment:
Could there be a reason why you used `id` instead of the original `encounterId`?

Reply:
No particular reason. I saw it done this way in the implementation of a different module (linked on the JIRA ticket) and used it. I didn't realise that the entities used `encounterId` and `visitId` as the PK field names, and the test passed so I didn't notice a discrepancy.

Will update it :)